lib/chess/board.ex

```diff
@@ -30,22 +30,53 @@ defmodule Chess.Board do
     board["#{file},#{rank}"]
   end
 
-  def move_piece(board, %{"from" => from, "to" => to}) do
-    [from_file, from_rank] = from
-    [to_file, to_rank] = to
-
+  def move_piece(board, %{
+    "from" => [from_file, from_rank],
+    "to" => [to_file, to_rank]}
+  ) do
     {piece, board} = Map.pop(board, "#{from_file},#{from_rank}")
     {piece_captured, board} = Map.pop(board, "#{to_file},#{to_rank}")
+    board = Map.put(board, "#{to_file},#{to_rank}", piece)
```

Comment:
You seem to be doing the `"#{file},#{rank}"` a few times here. What do you think about extracting it to a function somewhere? What does it represent?

Reply:
These strings are how the pieces are indexed in the board structure. I would rather use tuples or something here but I was limited by what I could serialize to JSON.